### PR TITLE
Move Terraform backend to S3 and deploy from master with GitHub Action

### PR DESCRIPTION
**Pull request by mogul**
Relates to https://github.com/GSA/datagov-deploy/issues/1878

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,58 +5,128 @@ services.
...
+| Secret Name | Description |
+|-------------|-------------|
+| AWS_ACCESS_KEY_ID | the S3 bucket key|
+| AWS_SECRET_ACCESS_KEY | the S3 bucket  |
+| BUCKET | the S3 bucket name |
+| TF_VAR_AWS_ACCESS_KEY_ID | the key for brokering resources in AWS |
+| TF_VAR_AWS_SECRET_ACCESS_KEY | the secret for brokering resources in AWS |
 
+Once these secrets are in place, any changes to the master branch will be
+deployed automatically.
```
Comment (adborden) [inline README.md:114]: The `---` is getting parsed as markdown heading syntax.
```suggestion
deployed automatically.

```

**Comment by mogul** [comment]
I made a bunch of commits to make the whole thing downstream configurable; can you take another look?

File: locals.tf [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        space = sn
+      }
+    ]
+  ])
+
+  spaces_in_orgs = {
+    for os in local.org_spaces : "${os.org}/${os.space}" => {
+      org   = os.org
+      space = os.space
+    }
```
Comment (adborden) [inline locals.tf:21]: omg, for loops!

Comment (mogul) [reply locals.tf:21]: i no rite!?

**Pull request merged**